include/SH3/system/input_config.hpp

```diff
@@ -0,0 +1,31 @@
+/** @file
+ *  Defines configuration variables for the @ref sh3::system::input_system.
+ */
+#ifndef SH3_SYSTEM_INPUT_CONFIG_HPP_INCLUDED
+#define SH3_SYSTEM_INPUT_CONFIG_HPP_INCLUDED
+
+/**
+ *  Whether the @ref sh3::system::input_system should provide timing support for inputs.
+ *  
+ *  With this enabled, the @ref sh3::system::input_system keeps track what fraction of a frame the actions were active for.
+ *  
+ * @note Doesn't work right with SDL timestamps.
+ *       I think the event times are not when the input *happened*, but when the event-loop *received* them, which is not useful to us.
+ */
+#ifdef DOXYGEN
+#define INPUT_PROVIDE_TIMING 0
```

Comment:
The timestamps would be used for `input_system::action_state::NormalizedTime`. Without this feature it'll return either `0` or `1`, with it, it could be any value in between, indicating for what fraction of a frame the key was pressed.
It's not a super important or even necessarily useful feature. You could let the character move for `distance * NormalizedTime(PRESSED)`. Without this feature, the character moves the whole distance, while with it, if it worked properly, it'd move corresponding to how long the button was pressed. Perhaps the only time it's even noticeable would be when the game lags.
For single-player we can remedy this effect by bounding the delta-time on updates, so the game itself slows down if a desired frametime cannot be reached.